From 506e0fb1af9627d3e33ff3b10b01ff6a93d4fd8b Mon Sep 17 00:00:00 2001
From: Mitch Curtis <mitch.curtis@qt.io>
Date: Sat, 5 Oct 2024 23:12:47 +0800
Subject: [PATCH 5/8] Revert "QQmlDelegateModel: fix delegates not being
 created in certain cases"

This reverts commit 6561344dd2d1ba69abe6edec4fe340b256da9e13. It needs
to be fixed in a different way.

Fixes: QTBUG-127340
Pick-to: 6.7 6.5
Change-Id: I8503b22a5257e0fb5ee11a1bdf83d3dcab4a600a
Reviewed-by: Richard Moe Gustavsen <richard.gustavsen@qt.io>
(cherry picked from commit 281f620ceea03e7a222d796ae0cca917a9778368)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../auto/qml/qqmldelegatemodel/CMakeLists.txt |   1 -
 .../auto/qml/qqmldelegatemodel/data/reset.qml |  28 ----
 .../data/resetInQAIMConstructor.qml           |  28 ----
 .../tst_qqmldelegatemodel.cpp                 | 135 ++----------------
 4 files changed, 10 insertions(+), 182 deletions(-)
 delete mode 100644 qtdeclarative/tests/auto/qml/qqmldelegatemodel/data/reset.qml
 delete mode 100644 qtdeclarative/tests/auto/qml/qqmldelegatemodel/data/resetInQAIMConstructor.qml

diff --git a/qtdeclarative/tests/auto/qml/qqmldelegatemodel/CMakeLists.txt b/qtdeclarative/tests/auto/qml/qqmldelegatemodel/CMakeLists.txt
index 966f5229df..8d8a90e0a7 100644
--- a/qtdeclarative/tests/auto/qml/qqmldelegatemodel/CMakeLists.txt
+++ b/qtdeclarative/tests/auto/qml/qqmldelegatemodel/CMakeLists.txt
@@ -29,7 +29,6 @@ qt_internal_add_test(tst_qqmldelegatemodel
         Qt::QmlModelsPrivate
         Qt::QmlPrivate
         Qt::Quick
-        Qt::QuickPrivate
         Qt::QuickTestUtilsPrivate
     TESTDATA ${test_data}
 )
diff --git a/qtdeclarative/tests/auto/qml/qqmldelegatemodel/data/reset.qml b/qtdeclarative/tests/auto/qml/qqmldelegatemodel/data/reset.qml
deleted file mode 100644
index 0fcd5e8afa..0000000000
--- a/qtdeclarative/tests/auto/qml/qqmldelegatemodel/data/reset.qml
+++ /dev/null
@@ -1,28 +0,0 @@
-import QtQuick
-import Test
-
-Window {
-    id: root
-    width: 200
-    height: 200
-
-    property alias listView: listView
-
-    ResettableModel {
-        id: resetModel
-    }
-
-    ListView {
-        id: listView
-        anchors.fill: parent
-        model: resetModel
-
-        delegate: Rectangle {
-            implicitWidth: 100
-            implicitHeight: 50
-            color: "olivedrab"
-
-            required property string display
-        }
-    }
-}
diff --git a/qtdeclarative/tests/auto/qml/qqmldelegatemodel/data/resetInQAIMConstructor.qml b/qtdeclarative/tests/auto/qml/qqmldelegatemodel/data/resetInQAIMConstructor.qml
deleted file mode 100644
index cb1f226737..0000000000
--- a/qtdeclarative/tests/auto/qml/qqmldelegatemodel/data/resetInQAIMConstructor.qml
+++ /dev/null
@@ -1,28 +0,0 @@
-import QtQuick
-import Test
-
-Window {
-    id: root
-    width: 200
-    height: 200
-
-    property alias listView: listView
-
-    ResetInConstructorModel {
-        id: resetInConstructorModel
-    }
-
-    ListView {
-        id: listView
-        anchors.fill: parent
-        model: resetInConstructorModel
-
-        delegate: Rectangle {
-            implicitWidth: 100
-            implicitHeight: 50
-            color: "olivedrab"
-
-            required property string display
-        }
-    }
-}
diff --git a/qtdeclarative/tests/auto/qml/qqmldelegatemodel/tst_qqmldelegatemodel.cpp b/qtdeclarative/tests/auto/qml/qqmldelegatemodel/tst_qqmldelegatemodel.cpp
index 3f08d8fc85..47d531e8e8 100644
--- a/qtdeclarative/tests/auto/qml/qqmldelegatemodel/tst_qqmldelegatemodel.cpp
+++ b/qtdeclarative/tests/auto/qml/qqmldelegatemodel/tst_qqmldelegatemodel.cpp
@@ -5,7 +5,6 @@
 #include <QtCore/qjsonobject.h>
 #include <QtCore/qsortfilterproxymodel.h>
 #include <QtCore/QConcatenateTablesProxyModel>
-#include <QtCore/qtimer.h>
 #include <QtGui/QStandardItemModel>
 #include <QtQml/qqmlcomponent.h>
 #include <QtQml/qqmlapplicationengine.h>
@@ -13,17 +12,11 @@
 #include <QtQmlModels/private/qqmllistmodel_p.h>
 #include <QtQuick/qquickview.h>
 #include <QtQuick/qquickitem.h>
-#include <QtQuick/private/qquickitemview_p_p.h>
-#include <QtQuick/private/qquicklistview_p.h>
-#include <QtQuickTest/quicktest.h>
 #include <QtQuickTestUtils/private/qmlutils_p.h>
-#include <QtQuickTestUtils/private/visualtestutils_p.h>
 #include <QtTest/QSignalSpy>
 
 #include <forward_list>
 
-using namespace QQuickVisualTestUtils;
-
 class tst_QQmlDelegateModel : public QQmlDataTest
 {
     Q_OBJECT
@@ -33,8 +26,6 @@ public:
 
 private slots:
     void resettingRolesRespected();
-    void resetInQAIMConstructor();
-    void reset();
     void valueWithoutCallingObjectFirst_data();
     void valueWithoutCallingObjectFirst();
     void qtbug_86017();
@@ -55,9 +46,16 @@ private slots:
     void proxyModelWithDelayedSourceModelInListView();
 };
 
-class BaseAbstractItemModel : public QAbstractItemModel
+class AbstractItemModel : public QAbstractItemModel
 {
+    Q_OBJECT
 public:
+    AbstractItemModel()
+    {
+        for (int i = 0; i < 3; ++i)
+            mValues.append(QString::fromLatin1("Item %1").arg(i));
+    }
+
     QModelIndex index(int row, int column, const QModelIndex &parent = QModelIndex()) const override
     {
         if (parent.isValid())
@@ -95,21 +93,10 @@ public:
         return mValues.at(index.row());
     }
 
-protected:
+private:
     QVector<QString> mValues;
 };
 
-class AbstractItemModel : public BaseAbstractItemModel
-{
-    Q_OBJECT
-public:
-    AbstractItemModel()
-    {
-        for (int i = 0; i < 3; ++i)
-            mValues.append(QString::fromLatin1("Item %1").arg(i));
-    }
-};
-
 tst_QQmlDelegateModel::tst_QQmlDelegateModel()
     : QQmlDataTest(QT_QMLTEST_DATADIR)
 {
@@ -168,109 +155,7 @@ void tst_QQmlDelegateModel::resettingRolesRespected()
     QObject *root = engine.rootObjects().constFirst();
     QVERIFY(!root->property("success").toBool());
     model->change();
-    QTRY_VERIFY_WITH_TIMEOUT(root->property("success").toBool(), 100);
-}
-
-class ResetInConstructorModel : public BaseAbstractItemModel
-{
-    Q_OBJECT
-    QML_ELEMENT
-
-public:
-    ResetInConstructorModel()
-    {
-        beginResetModel();
-        QTimer::singleShot(0, this, &ResetInConstructorModel::finishReset);
-    }
-
-private:
-    void finishReset()
-    {
-        mValues.append("First");
-        endResetModel();
-    }
-};
-
-void tst_QQmlDelegateModel::resetInQAIMConstructor()
-{
-    qmlRegisterTypesAndRevisions<ResetInConstructorModel>("Test", 1);
-
-    QQuickApplicationHelper helper(this, "resetInQAIMConstructor.qml");
-    QVERIFY2(helper.ready, helper.failureMessage());
-    QQuickWindow *window = helper.window;
-    window->show();
-    QVERIFY(QTest::qWaitForWindowExposed(window));
-
-    auto *listView = window->property("listView").value<QQuickListView *>();
-    QVERIFY(listView);
-    QTRY_VERIFY_WITH_TIMEOUT(listView->itemAtIndex(0), 100);
-    QQuickItem *firstDelegateItem = listView->itemAtIndex(0);
-    QVERIFY(firstDelegateItem);
-    QCOMPARE(firstDelegateItem->property("display").toString(), "First");
-}
-
-class ResettableModel : public BaseAbstractItemModel
-{
-    Q_OBJECT
-    QML_ELEMENT
-
-public:
-    ResettableModel()
-    {
-        mValues.append("First");
-    }
-
-    void callBeginResetModel()
-    {
-        beginResetModel();
-        mValues.clear();
-    }
-
-    void appendData()
-    {
-        mValues.append(QString::fromLatin1("Item %1").arg(mValues.size()));
-    }
-
-    void callEndResetModel()
-    {
-        endResetModel();
-    }
-};
-
-// Tests that everything works as expected when calling beginResetModel/endResetModel
-// after the QAIM subclass constructor has run.
-void tst_QQmlDelegateModel::reset()
-{
-    qmlRegisterTypesAndRevisions<ResettableModel>("Test", 1);
-
-    QQuickApplicationHelper helper(this, "reset.qml");
-    QVERIFY2(helper.ready, helper.failureMessage());
-    QQuickWindow *window = helper.window;
-    window->show();
-    QVERIFY(QTest::qWaitForWindowExposed(window));
-
-    auto *listView = window->property("listView").value<QQuickListView *>();
-    QVERIFY(listView);
-    QQuickItem *firstDelegateItem = listView->itemAtIndex(0);
-    QVERIFY(firstDelegateItem);
-    QCOMPARE(firstDelegateItem->property("display").toString(), "First");
-
-    const auto delegateModel = QQuickItemViewPrivate::get(listView)->model;
-    QSignalSpy rootIndexChangedSpy(delegateModel, SIGNAL(rootIndexChanged()));
-    QVERIFY(rootIndexChangedSpy.isValid());
-
-    auto *model = listView->model().value<ResettableModel *>();
-    model->callBeginResetModel();
-    model->appendData();
-    model->callEndResetModel();
-    // This is verifies that handleModelReset isn't called
-    // more than once during this process, since it unconditionally emits rootIndexChanged.
-    QCOMPARE(rootIndexChangedSpy.count(), 1);
-
-    QTRY_VERIFY_WITH_TIMEOUT(listView->itemAtIndex(0), 100);
-    firstDelegateItem = listView->itemAtIndex(0);
-    QVERIFY(firstDelegateItem);
-    QCOMPARE(firstDelegateItem->property("display").toString(), "Item 0");
+    QTRY_VERIFY(root->property("success").toBool());
 }
 
 void tst_QQmlDelegateModel::valueWithoutCallingObjectFirst_data()
-- 
2.48.1

